app/seeders/env_data/ldap_seeder.rb

```diff
@@ -0,0 +1,115 @@
+#-- copyright
+
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+module EnvData
+  class LdapSeeder < Seeder
+    def seed_data!
+      print_status '    ↳ Creating LDAP connection' do
+        Setting.seed_ldap.each do |name, options|
+          ldap = LdapAuthSource.find_or_initialize_by(name:)
+
+          print_ldap_status(ldap)
+          upsert_settings(ldap, options)
+          update_filters(ldap, options['groupfilter'])
+        end
+      end
+
+      print_status '    ↳ Synchronizing LDAP connections' do
+        LdapGroups::SynchronizationJob.perform_now
+      end
+    end
+
+    def applicable?
+      Setting.seed_ldap.present?
+    end
+
+    private
+
+    # rubocop:disable Metrics/AbcSize
+    def upsert_settings(ldap, options)
+      ldap.host = options['host']
+      ldap.port = options['port']
+
+      ldap.tls_mode = options['security']
+      ldap.verify_peer = ActiveRecord::Type::Boolean.new.deserialize options.fetch('tlsverify', true)
+      ldap.onthefly_register = ActiveRecord::Type::Boolean.new.deserialize options.fetch('sync_users', false)
+      ldap.tls_certificate_string = options['certificate'].presence
+
+      ldap.filter_string = options['filter'].presence
+      ldap.base_dn = options['basedn']
+      ldap.account = options['binduser']
+      ldap.account_password = options['bindpassword']
+
+      ldap.attr_login = options['login_mapping'].presence
+      ldap.attr_firstname = options['firstname_mapping'].presence
+      ldap.attr_lastname = options['lastname_mapping'].presence
+      ldap.attr_mail = options['mail_mapping'].presence
+
+      ldap.save!
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def update_filters(ldap, filters)
+      return if filters.blank?
```

Comment:
This means that I cannot remove the last filter by this method. I can however, after having removed the env vars, remove the filter via the UI. This might be an edge case anyway. But it wouldn't be hard to change this into 

```
return if filters.blank? || ::LdapGroups::SynchronizedFilter.exists?(ldap_auth_source: ldap)
```

which would allow for this scenario at the cost of potentially removing a filter without wanting to if the LDAP ENV itself are set but the filter ENVs vor it are forgotten. I think that to be unlikely, though.

Reply:
Good catch. The condition needs to be inverted however. If the filter exists, we want to remove it.